relay/server/peer.go

```diff
@@ -162,22 +172,30 @@ func (p *Peer) healthcheck(ctx context.Context, hc *healthcheck.Sender) {
 }
 
 func (p *Peer) handleTransportMsg(msg []byte) {
-	peerID, err := messages.UnmarshalTransportID(msg)
+	version, peerID, err := messages.UnmarshalTransportID(msg)
```

Comment:
Already checked the version in the read loop based on the same information. I think here enough to handle the peerID and payload.